realm/realm-library/src/main/java/io/realm/RealmObjectSchema.java

```diff
@@ -462,6 +462,22 @@ public void setEmbedded(boolean embedded) {
         }
     }
 
+    /**
+     * Retrieves the type of a given property.
+     *
+     * @param property the field for which we want to know the type.
+     * @return the name of the property.
+     * @throws IllegalStateException if the given property is not found in the schema.
+     */
+    public String getPropertyType(String property) {
+        ColumnInfo.ColumnDetails columnDetails = columnInfo.getColumnDetails(property);
+        if (columnDetails == null) {
+            throw new IllegalStateException(String.format("Property %s not found.", property));
+        } else {
+            return columnDetails.linkedClassName;
```

Comment:
What is returned if the property is not a Link or List? ... Right now `getPropertyType` sounds similar to `getFieldType()` which returns the type of the Field... maybe rename this to `getLinkedType()` which will throw an illegal state exception with something like "Field does not contain a link, but X" kind of error if the field doesn't contain a link